### PR TITLE
fix: Karma reported test duration is negative fix #143

### LINKED ISSUE #143: Karma reported test duration is negative
[Originally reported](https://github.com/karma-runner/karma/issues/2425#issuecomment-263017704) to the Karma project.

### Expected behaviour

Running karma tests takes ~12 seconds, expect test duration (runtime) to be ~12 seconds.
### Actual behaviour

PhantomJS 2.1.1 (Windows 7 0.0.0): Executed 316 of 316 SUCCESS (1.959 secs / **-41118 mins 50.274 secs**)
TOTAL: 316 SUCCESS
### Environment Details

[Angular Yeoman](https://github.com/yeoman/generator-angular) Project.
Windows 7
- Karma version (output of `karma --version`): 1.3.0
- Relevant part of your `karma.config.js` file
### Steps to reproduce the behaviour
1. `grunt test`
2. observe large negative test duration.

I think I found the issue, one of the tests is using `jasmine.clock().mocktime(time)` and that seems to be screwing up how Karma reports the duration for the test.  Since the mocked time is far enough in the past I'm guessing that the signed integer is getting flipped and going way negative.

### PR DESCRIPTION
**Pull request by maksimr**

### CODE REVIEW

**Comment by maksimr** [comment]
\cc @dignifiedquire @reustonium

File: src/adapter.js [hunk 1]
```diff
@@ -157,6 +157,7 @@ function getAllSpecNames (topSuite) {
  */
 function KarmaReporter (tc, jasmineEnv) {
   var currentSuite = new SuiteNode()
+  var _Date = Date
```
Comment (dignifiedquire) [inline src/adapter.js]: Please add a comment as to why this is happening, otherwise the next one editing this code will remove it :D

**Comment by dignifiedquire** [comment]
Thanks @maksimr, one minor nitpick

**Comment by dignifiedquire** [comment]
Thanks :octocat:

**Comment by reustonium** [comment]
Thanks for the fix!

**Pull request merged**